service/history/task/transfer_active_task_executor.go

```diff
@@ -1696,11 +1698,12 @@ func (t *transferActiveTaskExecutor) processParentClosePolicy(
 			})
 	}
 
-	for _, childInfo := range sameClusterChildInfos {
+	for initiatedID, childDomainID := range sameClusterChildDomainIDs {
+		childInfo := childInfos[initiatedID]
```

Comment:
I thought keeping the child info in the value field was just a reference. Was it taking a full copy of childInfo? If so, we could just use the reference to the value to keep here simpler to avoid querying the info with eventID again?

Reply:
yes, childInfo is a reference. I changed the type of `sameClusterChildXXX` to record the child domainID, so that we don't need to query domain cache again here. 
I considered storing both information, which requires either defining a new type to hold both domainID and childInfo, or use two slices and match them by index. Both approaches kinda make the code more complicated so I ended up using this approach. 
What's your thought on this? Happy to change the implementation.

Reply:
I see, this looks good to me. I don't have a strong opinion.